src/System.Private.CoreLib/src/System/Threading/Timer.Unix.cs

```diff
@@ -13,109 +14,141 @@ namespace System.Threading
     //
     // Unix-specific implementation of Timer
     //
-    internal partial class TimerQueue
+    internal partial class TimerQueue : IThreadPoolWorkItem
     {
+        private static List<ScheduledTimer> s_scheduledTimers;
+        private static List<int> s_timerIDsToFire;
+
         /// <summary>
         /// This event is used by the timer thread to wait for timer expiration. It is also
         /// used to notify the timer thread that a new timer has been set.
         /// </summary>
-        private AutoResetEvent _timerEvent;
+        private static readonly AutoResetEvent s_timerEvent = new AutoResetEvent(false);
 
-        /// <summary>
-        /// This field stores the value of next timer that the timer thread should install.
-        /// </summary>
-        private volatile int _nextTimerDuration;
+        private readonly int _id;
 
         private TimerQueue(int id)
         {
+            _id = id;
         }
```

Comment:
`_id` and the constructor can now be moved to the shared code in Timer.cs. It is identical for all implementations.